bitbots_odometry/src/odometry_fuser.cpp

```diff
@@ -114,6 +141,47 @@ OdometryFuser::OdometryFuser() {
   }
 }
 
+tf2::Transform OdometryFuser::getCurrentRotationPoint() {
+  geometry_msgs::TransformStamped rotation_point;
+  tf2::Transform rotation_point_tf;
+
+  // if center of pressure is available, it is the point of rotation
+  try {
+    rotation_point = tf_buffer_.lookupTransform("/base_link", "/cop", ros::Time(0));
+  } catch (tf2::TransformException ex) {
+    // otheriwse point of rotation is current support foot sole or center point of the soles if double support
+    if (current_support_state_ == 'r') {
+      try {
+        rotation_point = tf_buffer_.lookupTransform("/base_link", "/l_sole", ros::Time::now());
+      } catch (tf2::TransformException ex) {
+        ROS_ERROR("%s", ex.what());
+      }
+    } else if (current_support_state_ == 'l') {
+      try {
+        rotation_point = tf_buffer_.lookupTransform("/base_link", "/r_sole", ros::Time::now());
+      } catch (tf2::TransformException ex) {
+        ROS_ERROR("%s", ex.what());
+      }
+    } else if (current_support_state_ == 'd') {
+      // use point between soles
+      geometry_msgs::TransformStamped base_to_l_sole;
+      base_to_l_sole = tf_buffer_.lookupTransform("/base_link", "/l_sole", ros::Time::now());
+      geometry_msgs::TransformStamped l_to_r_sole;
+      l_to_r_sole = tf_buffer_.lookupTransform("/l_sole", "/r_sole", ros::Time::now());
+      tf2::Transform base_to_l_sole_tf;
+      tf2::fromMsg(base_to_l_sole, base_to_l_sole_tf);
+      tf2::Transform l_to_r_sole_tf;
+      tf2::fromMsg(l_to_r_sole, l_to_r_sole_tf);
+
+      //todo
+      //rotation_point = base_to_l_sole_tf * (0.5 * l_to_r_sole_tf);
+    } else {
+      ROS_ERROR("unknwn support state %c", current_support_state_);
+    }
+  }
```

Comment:
if I see it correctly the whole if block could be dropped and simplified to:

```
if (current_support_state_ != 'r' || current_support_state_ != 'l' || current_support_state_ != 'd') {
      ROS_ERROR("unknwn support state %c", current_support_state_);
    } else {
      if (current_support_state_ == 'd') {
        // use point between soles
	geometry_msgs::TransformStamped base_to_l_sole;
	base_to_l_sole = tf_buffer_.lookupTransform("/base_link", "/l_sole", ros::Time::now());
        geometry_msgs::TransformStamped l_to_r_sole;
        l_to_r_sole = tf_buffer_.lookupTransform("/l_sole", "/r_sole", ros::Time::now());
        tf2::Transform base_to_l_sole_tf;
        tf2::fromMsg(base_to_l_sole, base_to_l_sole_tf);
        tf2::Transform l_to_r_sole_tf;
        tf2::fromMsg(l_to_r_sole, l_to_r_sole_tf);

        //todo
        //rotation_point = base_to_l_sole_tf * (0.5 * l_to_r_sole_tf);
      } else {
        // otheriwse point of rotation is current support foot sole or center point of the soles if double support
	char foo = 'r';
        if (current_support_state_ == 'r') { foo = 'l'; }
        try {
          rotation_point = tf_buffer_.lookupTransform("/base_link", "/"+ foo + "_sole", ros::Time::now());
        } catch (tf2::TransformException ex) {
          ROS_ERROR("%s", ex.what());
        }
      }
    }
  }
```

could be maybe even more simplified.

Reply:
we changed it a bit, but not exactly as you proposed, since we think its easier to have the error at the end, if none of the cases was true

Reply:
Yes, I see but be aware before you have `"/r_sole"` on `current_support_state_ == 'l'` and `"/l_sole"` on `current_support_state_ == 'r'` but now you will have `"/r_sole"` on `current_support_state_ == 'r'` and `"/l_sole"` on `current_support_state_ == 'l'` that's why I did it as above. If it doesn't matter ignore this comment and for me it will be resolved.

Reply:
that was actually wrong, we corrected it

Reply:
but thanks for the heads up :)